core/src/main/java/org/ldaptive/handler/RecursiveResultHandler.java

```diff
@@ -196,19 +203,18 @@ private void readSearchAttribute(final LdapEntry entry, final List<String> searc
    * @param  dn  to get attribute(s) for
    * @param  entry  to merge with
    * @param  searchedDns  list of DNs that have been searched for
+   *
+   * @throws  LdapException  if an error occurs performing a search
    */
   private void recursiveSearch(final String dn, final LdapEntry entry, final List<String> searchedDns)
+    throws LdapException
   {
     if (!searchedDns.contains(dn)) {
 
       LdapEntry newEntry = null;
-      try {
-        final SearchResponse result = performSearch(dn, retAttrs);
-        if (result.isSuccess() && result.entrySize() == 1) {
-          newEntry = result.getEntry();
-        }
-      } catch (LdapException e) {
-        logger.warn("Error retrieving attribute(s): {} for dn {}", Arrays.toString(retAttrs), dn, e);
+      final SearchResponse result = performSearch(dn, retAttrs);
+      if (result.isSuccess() && result.entrySize() == 1) {
+        newEntry = LdapEntry.copy(result.getEntry());
```

Comment:
I don't see the value of making a copy of the result here.

Reply:
The entry gets mutated below, you need a copy in order to change the entry coming from a search operation.